src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3747,7 +3747,12 @@ void addToStringMethod(MethodSymbol printMethod)
                 {
                     if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
                     {
-                        var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
+                        var toStringMethod = new SynthesizedRecordToString(
```

Comment:
It occurs to me that it should be specified that the synthesized `ToString` has the same `readonly`-ness as the `PrintMembers`, whether it is synthesized or explicitly declared.

Reply:
Have made the change.